### PR TITLE
parser.y: Always populate SelectField offset

### PR DESCRIPTION
**Pull request by kyleconroy**
👋 This is my time contributing code to this project, so please let me know if I need to do more work to get this approved. I'm happy to add tests, documentation, etc.

# What problem does this PR solve?
This is a partial solution to #910 

### What is changed and how it works?
`ast.SelectField` already has an Offset field. I've updated parser.y to populate this field whenever an `ast.SelectField` is created.

### CODE REVIEW

**Comment by kyleconroy** [comment]
Any feedback for this pull request? Is this something you'd be interested in merging if I fixed the test failures?

File: parser.y [hunk 1]
```diff
@@ -4755,23 +4755,36 @@ LikeEscapeOpt:
 Field:
 	'*'
 	{
-		$$ = &ast.SelectField{WildCard: &ast.WildCardField{}}
+		$$ = &ast.SelectField{
+			WildCard: &ast.WildCardField{},
+			Offset:   parser.startOffset(&yyS[yypt]),
```
Comment (tangenta) [inline parser.y:4760]: The offset field has been initialized here:

https://github.com/pingcap/parser/blob/12de604100a41a10651cd9083a16565284813c40/parser.y#L4845-L4851

What's the purpose to make this change? Are there any tests or use cases?

**Comment by tangenta** [comment]
> Any feedback for this pull request? Is this something you'd be interested in merging if I fixed the test failures?

Sorry for the late reply. I thought this is still WIP..

File: parser.y (hunk 1 above)
Comment (kyleconroy) [reply parser.y:4760]: Hmm, if this is being populated I'm not sure why I needed to change parser.y at all. I'll need to investigate why it wasn't working for me.

**Comment by kyleconroy** [comment]
> Sorry for the late reply. I thought this is still WIP..

No worries! I was hoping to get feedback before I did too much work. Thanks for explaining how Offset is currently being populated, as I think it makes this PR unnecessary.

**Comment by kyleconroy** [comment]
I finally got around to testing this; the current behavior works perfectly.

**Pull request closed without merging**